File: Lab/Lab_1_2/src/Lab_1/ReadFile.py
import os
from pathlib import Path


class ReadFile:

    @staticmethod
    def file_exists(file_path: str) -> bool:
        path = Path(file_path)
        return path.exists() and path.is_file()

    @staticmethod
    def is_file_empty(file_path: str) -> bool:
        if not file_path.endswith(".txt"):
            raise ValueError(f"Only Txt File: {file_path}")
        return os.path.getsize(file_path) == 0

    @staticmethod
    def read_whole_file(file_path: str) -> str:
        with open(file_path, "r", encoding="utf-8") as f:
            return f.read()

    @staticmethod
    def print_file_content(file_path: str):
        if not ReadFile.file_exists(file_path):
            print(f"Can't Find File: {file_path}")
            return

        try:
            if ReadFile.is_file_empty(file_path):
                print(f"Empty File: {file_path}")
                return

            content = ReadFile.read_whole_file(file_path)
            return content.split(" ")  # list of strings

        except Exception as e:
            print(f"Can't Read File: {e}")

    @staticmethod
    def first_main():
        file_path = input("Insert File Path: ")
        str_list = ReadFile.print_file_content(file_path)
        int_list = [int(x) for x in str_list]
        return int_list


if __name__ == "__main__":
    print(ReadFile.first_main())

File: Lab/Lab_1_1/src/main/java/ip/group/Lab_1/ReadFile.java
package ip.group.Lab_1;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.Scanner;

public class ReadFile {

    public static boolean fileExists(String filePath) {
        Path path = Paths.get(filePath);
        return Files.exists(path) && Files.isRegularFile(path);
    }

    public static boolean isFileEmpty(String filePath) {
        if (!filePath.endsWith(".txt")) {
            throw new IllegalArgumentException("Only Txt File: " + filePath);
        }

        try {
            return Files.size(Paths.get(filePath)) == 0;
        } catch (IOException e) {
            throw new RuntimeException("Can't Read File Size: " + e.getMessage(), e);
        }
    }

    public static String readWholeFile(String filePath) throws IOException {
        return Files.readString(Paths.get(filePath));
    }

    public static String printFileContent(String filePath) {
        if (!fileExists(filePath)) {
            System.out.println("Can't Find File: " + filePath);
            return null;
        }

        if (isFileEmpty(filePath)) {
            System.out.println("Empty File: " + filePath);
            return null;
        }

        try {
            String content = readWholeFile(filePath);
            return content;
        } catch (IOException e) {
            System.out.println("Read File Error: " + e.getMessage());
        }
        return null;
    }

    public static int[] toNumberSplit(String str) {
        return Arrays.stream(str.trim().split("\\s+")).mapToInt(Integer::parseInt).toArray();
    }

    public static String fistMain() {
        Scanner scan = new Scanner(System.in);

        System.out.println("Insert File Path:");
        String filePath = scan.nextLine();

        String str = printFileContent(filePath);
        return str;
    }


    public static void main(String[] args) {
        ReadFile readFile = new ReadFile();
        System.out.println(Arrays.toString(toNumberSplit(fistMain())));
    }
}
